pkg/operator/ceph/spec/spec.go

```diff
@@ -206,6 +211,7 @@ func DaemonVolumeMounts(dataPaths *config.DataPathMap, keyringResourceName strin
 	if dataPaths.HostLogDir != "" {
 		// logs are not persisted to host, so no mount is needed
 		mounts = append(mounts, StoredLogVolumeMount())
+		mounts = append(mounts, StoredCrashVolumeMount())
```

Comment:
I am concerned that gating the crash mount is gated behind existence of the log dir. What I am seeing is that `HostLogDir` and `HostCrashDir` are related and that they are set (or not set) together as a pair. It would be better to have both of them only use one field together in that case.

Why not change the `DataPathMap` to instead of `HostLogDir` and `HostCrashDir` keep the parent dir set as `HostLogAndCrashDir` and then add methods onto `DataPathMap` for `HostLogDir` which will return the subdirs?

```go
// A DataPathMap is a struct which contains information about where Ceph daemon data is stored in
// containers and whether the data should be persisted to the host. If it is persisted to the host,
// directory on the host where the specific daemon's data is stored is given.
type DataPathMap struct {
	HostDataDir string
	ContainerDataDir string
	HostLogAndCrashDir string
}

func (d *DataPathMap) HostLogDir() string {
	return strings.Join(d.HostLogAndCrashDir, "log")
}

func (d * DataPathMap) HostCrashDir() string {
	return strings.Join(d.HostLogAndCrashDir, "crash")
}
```

Reply:
I've started looking into this in more details and so far my results:

Expending the struct is nice, however, in practice, this means passing the struct to a lot of places  (`PodVolumes()` to get `dataPaths.HostLogDir(namespace)` and more), as well as exposing the namespace. This really makes a lot of changes everywhere so I'm not really sure if that's worth it. Simply relying on constant/var seems simpler although vars could be changed I have to admit... I don't see this as a high risk anyway, but implementing as you requested really implies a lot of changes and reduces the code simplicity here.

What's your view on that?

Reply:
I don't see clearly how what I've said requires substantially more changes. Anywhere the additional struct fields are used already has access to the struct, and what I have requested is to add methods to the struct instead of fields when those fields are always filled in with constant values. I don't see how usage is any different.